presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ProjectNode.java

```diff
@@ -74,18 +71,6 @@ public PlanNode getSource()
         return source;
     }
 
-    public boolean isIdentity()
```

Comment:
After checking the file again, I think the right to do this (after migrating `Expression` to `RowExpression` for `Assignment`) is:
```java
public boolean isIdentity()	
{	
    for (Map.Entry<Symbol, RowExpression> entry : assignments.entrySet()) {	
        RowExpression expression = entry.getValue();	
        Symbol symbol = entry.getKey();	
        if (!(expression instanceof VariableReferenceExpression && ((VariableReferenceExpression) expression).getName().equals(symbol.getName()))) {	
            return false;	
        }	
    }	
    return true;	
}

Reply:
I dont completly agree. I think we should support both OriginalExpression (down cast) and VariableReferenceExpression since this is called from both rules and cost rules.

Reply:
Fair enough